task/upload.go

```diff
@@ -367,11 +383,21 @@ func complementCatalystPipeline(tctx *TaskContext, assetSpec api.AssetSpec, call
 }
 
 func assetOutputLocations(tctx *TaskContext) ([]OutputName, []clients.OutputLocation, error) {
-	outputNames, outputLocations, err := outputLocations(tctx.OutputOSObj.URL, tctx.OutputAsset.PlaybackID)
+	playbackId := tctx.OutputAsset.PlaybackID
+	outputNames, outputLocations, err := outputLocations(tctx.OutputOSObj.URL, playbackId)
 	if err != nil {
 		return nil, nil, err
 	}
-
+	mp4Url, _ := url.JoinPath(mp4FileName(playbackId))
```

Comment:
You're using the wrong `url.JoinPath` function here, and is not sending a full URL to the `output_location` here. You must parse the object store URL like on line 420 🌿.

Reply:
Yeah I was looking into that :eyes: It was importing `url` as the package, because outside of the outputLocations function I wasn't parsing the outUrl. Crazy to me that I can define a variable in a scope with the same name of a global package without go complaining, btw

Reply:
Yeah that's really confusing haha
Fell for that a couple of times which is how I spotted it more easily here